### PR TITLE
Add FormObject QuestionForm, QuestionAnswersForm

### PR DESCRIPTION
**Pull request by OmegaLord**

### CODE REVIEW

File: app/forms/base_form.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  include ActiveModel::Model
+  include ActiveModel::Serialization
+
+  delegate :attributes=, :attributes, to: :attachments, prefix: true
+  delegate :valid?, to: :object
+  delegate :to_param, :errors, :persisted?, to: :object
+
+  attr_accessor :object, :current_user
+
+  def initialize(object, user, params = {})
```
Comment (AZatsepa) [inline app/forms/base_form.rb]: `object, params = {}` (params.merge(user: current_user)

File: app/forms/answer_form.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class AnswerForm < BaseForm
+  alias answer object
```
Comment (AZatsepa) [inline app/forms/answer_form.rb]: Is unnecessary

File: app/forms/answer_form.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  validates :body, presence: true
+
+  delegate :id, :body, :attachments, :question, :user, :user_id, :question_id, :comments, to: :answer
+
+  def self.model_name
+    ActiveModel::Name.new(self, nil, 'Answer')
+  end
+
```
Comment (AZatsepa) [inline app/forms/answer_form.rb]: delegate to answer

File: app/forms/base_form.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
+class BaseForm
+  extend ActiveModel::Naming
+  include ActiveModel::Model
+  include ActiveModel::Serialization
+
+  delegate :attributes=, :attributes, to: :attachments, prefix: true
+  delegate :valid?, to: :object
```
Comment (AZatsepa) [inline app/forms/base_form.rb]: Create validation in a form

File: app/forms/base_form.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
+class BaseForm
+  extend ActiveModel::Naming
+  include ActiveModel::Model
+  include ActiveModel::Serialization
+
+  delegate :attributes=, :attributes, to: :attachments, prefix: true
```
Comment (AZatsepa) [inline app/forms/base_form.rb]: ???

File: app/forms/base_form.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    @current_user = user
+    @attachments_params = uploaded_attachments params
+    @params = params
+  end
+
+  def save
+    return false unless valid?
+
+    object.save
+    save_attachments if attachments_params
```
Comment (AZatsepa) [inline app/forms/base_form.rb]: `if attachments_params && object.save`

File: app/forms/base_form.rb [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  def update
+    return false unless current_user == object.user
+
+    object.update params
+    save_attachments if attachments_params
+    true
```
Comment (AZatsepa) [inline app/forms/base_form.rb]: ???

File: app/forms/base_form.rb [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      if attachment_param['_destroy'].to_i == 1
+        Attachment.find(attachment_param['id']).destroy
+      elsif attachment_param['file']
+        Attachment.create(attachment_param.merge(attachmentable: object))
+      end
+    end
+  end
+
+  def uploaded_attachments(params)
+    # binding.pry
```
Comment (AZatsepa) [inline app/forms/base_form.rb]: Remove it

File: app/forms/question_form.rb [hunk 9]
```diff
@@ -0,0 +1,19 @@
+class QuestionForm < BaseForm
+  alias question object
```
Comment (AZatsepa) [inline app/forms/question_form.rb]: Is unnecessary

File: app/forms/question_form.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+  validates :body, :title, presence: true
+
+  delegate :id, :body, :title, :attachments, :user, :comments, to: :question
```
Comment (AZatsepa) [inline app/forms/question_form.rb]: delegate to object

File: app/forms/base_form.rb (hunk 7 above)
Comment (OmegaLord) [reply app/forms/base_form.rb]: Need to return something like `true`, because `save_attachments` return `[]`.
Now I return `object`.

**Pull request merged**